tests/test_parsing/test_file.py

```diff
@@ -0,0 +1,501 @@
+    """))
+    assert parsed == {
+        REPO_PYPI: get_reqs('othername >=5')
+    }
+
+
+def test_comma_separated_import():
+    parsed = parsing._parse_content(io.StringIO("""
+        from foo import bar, baz, qux   # fades
+    """))
+    assert parsed == {
+        REPO_PYPI: get_reqs('foo')
+    }
+
+
+def test_other_lines_with_fades_string():
+    parsed = parsing._parse_content(io.StringIO("""
+        import bar # fades
+        print("screen fades to black")
+    """))
+    assert parsed == {
+        REPO_PYPI: get_reqs('bar')
+    }
+
+
+def test_commented_line(logged):
+    parsed = parsing._parse_content(io.StringIO("""
+        #import foo   # fades
+    """))
+    assert parsed == {}
+    logged.assert_not_warning("Not understood fades")
+
+
+def test_with_fades_commented_line(logged):
+    parsed = parsing._parse_content(io.StringIO("""
+        #import foo   # fades
+        import bar   # fades
+    """))
+    assert parsed == {
+        REPO_PYPI: get_reqs('bar')
+    }
+    logged.assert_not_warning("Not understood fades")
+
+
+def test_with_commented_line(logged):
+    parsed = parsing._parse_content(io.StringIO("""
+        import bar   # fades
+        # a commented line
+    """))
+    assert parsed == {
+        REPO_PYPI: get_reqs('bar')
+    }
+    logged.assert_not_warning("Not understood fades")
+
+
+def test_vcs_explicit():
+    parsed = parsing._parse_content(io.StringIO("""
+        import foo    # fades vcs::superurl
+    """))
+    assert parsed == {
+        REPO_VCS: [parsing.VCSDependency('superurl')]
+    }
+
+
+def test_vcs_implicit():
+    parsed = parsing._parse_content(io.StringIO("""
+        import foo    # fades   http://www.whatever/project
+    """))
+    assert parsed == {
+        REPO_VCS: [parsing.VCSDependency('http://www.whatever/project')]
+    }
+
+
+def test_mixed():
+    parsed = parsing._parse_content(io.StringIO("""
+        import foo    # fades vcs::superurl
+        import bar    # fades
+    """))
+    assert parsed == {
+        REPO_VCS: [parsing.VCSDependency('superurl')],
+        REPO_PYPI: get_reqs('bar'),
+    }
+
+
+def test_fades_and_hashtag_mentioned_in_code():
+    """Test the case where a string contains both: fades and hashtag (#)
+    but is not an import.
+    """
+    parsed = parsing._parse_content(io.StringIO("""
+      'http://fades.readthedocs.io/en/release-7-0/readme.html#how-to-use-it'
+    """))
+    assert parsed == {}
+
+
+def test_fades_and_hashtag_mentioned_in_code_mixed_with_imports():
+    parsed = parsing._parse_content(io.StringIO("""import requests  # fades
+
+      'http://fades.readthedocs.io/en/release-7-0/readme.html#how-to-use-it'
+    """))
+    assert parsed == {
+        REPO_PYPI: get_reqs('requests')
+    }
+
+
+def test_fades_user_strange_comment_with_hashtag_ignored():
+    parsed = parsing._parse_content(io.StringIO("""
+      import foo # fades==2 # Some comment with #hashtash
+    """))  # noqa
```

Comment:
May this `# noqa` be removed? Just try it, it looks it's left there from ancient times